app/models/developer.rb

```diff
@@ -74,8 +76,8 @@ class Developer < ApplicationRecord
   scope :featured, -> { where("featured_at >= ?", FEATURE_LENGTH.ago).order(featured_at: :desc) }
   scope :newest_first, -> { order(created_at: :desc) }
   scope :profile_reminder_notifications, -> { where(profile_reminder_notifications: true) }
-  scope :recently_active, -> { where("developers.updated_at >= ?", RECENTLY_ACTIVE_LENGTH.ago) }
-  scope :source_contributor, -> { where("source_contributor >= ?", true) }
+  scope :recently_active, -> { joins(:badge).where(developers_badges: {recently_active: true}) }
```

Comment:
Let's create a scope on the badge model to check for recently active (see my comment above about storing the timestamp).